das/factory.go

```diff
@@ -148,3 +188,201 @@ func CreateBatchPosterDAS(
 
 	return daWriter, daReader, &lifecycleManager, nil
 }
+
+func CreateDAComponentsForDaserver(
+	ctx context.Context,
+	config *DataAvailabilityConfig,
+	l1Reader *headerreader.HeaderReader,
+	seqInboxAddress *common.Address,
+) (DataAvailabilityServiceReader, DataAvailabilityServiceWriter, DataAvailabilityServiceHealthChecker, *LifecycleManager, error) {
+	if !config.Enable {
+		return nil, nil, nil, nil, nil
+	}
+
+	// Check config requirements
+	if config.AggregatorConfig.Enable {
+		return nil, nil, nil, nil, errors.New("--data-availability.rpc-aggregator")
```

Comment:
Should this error be a bit more specific?

Reply:
It turns out that check is not needed because the aggregator config is only added for nitro nodes, not `daserver`